src/pages/patientView/PatientViewPage.tsx

```diff
@@ -85,19 +88,44 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
 
     @observable private mutationTableColumnVisibility: {[columnId: string]: boolean}|undefined;
     @observable private cnaTableColumnVisibility: {[columnId: string]: boolean}|undefined;
+    
+    // use this wrapper rather than interacting with the url directly 
+    private urlWrapper: PatientViewUrlWrapper
 
     constructor(props: IPatientViewPageProps) {
 
         super(props);
+        this.urlWrapper = new PatientViewUrlWrapper(props.routing);
 
+        if (
+            this.props.routing.location.pathname.includes("/" + PagePath.Patient) &&
+            this.urlWrapper.query.studyId
+        ) {
+            trackPatient(this.urlWrapper.query.studyId);
```

Comment:
i'm guessing you call trackPatient here because the reaction below only fires on subsequent changes?  If that's the case, look at option fireImmediately that reaction takes (https://mobx.js.org/refguide/reaction.html).